benchmark/src/main/java/com.google.cloud.bigquery/ConnImplBenchmark.java

```diff
@@ -0,0 +1,181 @@
+import java.util.logging.Level;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+@Fork(value = 1)
+@BenchmarkMode(Mode.AverageTime)
+@Warmup(iterations = 0)
+@Measurement(iterations = 3)
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class ConnImplBenchmark {
+  @Param({"500000", "1000000", "10000000", "100000000"}) // 500K, 1M, 10M, and 100M
+  public int rowLimit;
+
+  private ConnectionSettings connectionSettingsReadAPIEnabled, connectionSettingsReadAPIDisabled;
+  private long numBuffRows = 100000L;
+  private final String DATASET = "bigquery_test_dataset";
+  private final String QUERY = "SELECT * FROM tlc_yellow_trips_2017_stephwang LIMIT %s";
+
+  @Setup
+  public void setUp() throws IOException {
+    java.util.logging.Logger.getGlobal().setLevel(Level.ALL);
+    ReadClientConnectionConfiguration clientConnectionConfiguration;
+
+    clientConnectionConfiguration =
+        ReadClientConnectionConfiguration.newBuilder()
+            .setTotalToPageRowCountRatio(10L)
+            .setMinResultSize(200000L)
+            .setBufferSize(numBuffRows)
+            .build();
+
+    connectionSettingsReadAPIEnabled =
+        ConnectionSettings.newBuilder()
+            .setDefaultDataset(DatasetId.of(DATASET))
+            .setNumBufferedRows(numBuffRows) // page size
+            .setPriority(
+                QueryJobConfiguration.Priority
+                    .INTERACTIVE) // DEFAULT VALUE - so that isFastQuerySupported returns false
+            .setReadClientConnectionConfiguration(clientConnectionConfiguration)
+            .setUseReadAPI(true) // enable read api
+            .build();
+    connectionSettingsReadAPIDisabled =
+        ConnectionSettings.newBuilder()
+            .setDefaultDataset(DatasetId.of(DATASET))
+            .setNumBufferedRows(numBuffRows) // page size
+            .setPriority(
+                QueryJobConfiguration.Priority
+                    .INTERACTIVE) // so that isFastQuerySupported returns false
+            .setReadClientConnectionConfiguration(clientConnectionConfiguration)
+            .setUseReadAPI(false) // disable read api
+            .build();
+  }
+
+  @Benchmark
+  public void iterateRecordsUsingReadAPI(Blackhole blackhole)
+      throws InterruptedException, BigQuerySQLException {
+    Connection connectionReadAPIEnabled =
```

Comment:
+1 It's a good suggestion to moved connection creation logic out since we care only about query performance here (and hopefully this could even improve the benchmark results). @prash-mi

Reply:
We can definitely try excluding the connection initialisation times using the technique mentioned, I will update the benchmarks

Reply:
Let's do this in a follow-up PR.

Reply:
https://github.com/googleapis/java-bigquery/issues/2015 to track this work.